api/v1alpha1/helmreleaseproxy_webhook.go

```diff
@@ -58,20 +59,23 @@ func (p *HelmReleaseProxy) Default() {
 
 var _ webhook.Validator = &HelmReleaseProxy{}
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type.
 func (r *HelmReleaseProxy) ValidateCreate() (admission.Warnings, error) {
 	helmreleaseproxylog.Info("validate create", "name", r.Name)
 
 	// TODO(user): fill in your validation logic upon object creation.
 	return nil, nil
 }
 
-// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
 func (r *HelmReleaseProxy) ValidateUpdate(oldRaw runtime.Object) (admission.Warnings, error) {
 	helmreleaseproxylog.Info("validate update", "name", r.Name)
 
 	var allErrs field.ErrorList
-	old := oldRaw.(*HelmReleaseProxy)
+	old, ok := oldRaw.(*HelmReleaseProxy)
+	if !ok {
+		return nil, fmt.Errorf("cannot convert object to HelmReleaseProxy type")
```

Comment:
```suggestion
		return nil, apierrors.NewBadRequest(fmt.Sprintf("expected a HelmReleaseProxy but got a %T", old))
```

where we have the import `apierrors "k8s.io/apimachinery/pkg/api/errors"` based off of the CAPI webhooks.